cmd/gator/test/test.go

```diff
@@ -96,31 +102,68 @@ func run(cmd *cobra.Command, args []string) {
 	os.Exit(exitCode)
 }
 
-func formatOutput(flagOutput string, results []*test.GatorResult) string {
+func formatOutput(flagOutput string, results []*test.GatorResult, stats []*instrumentation.StatsEntry) string {
 	switch strings.ToLower(flagOutput) {
 	case stringJSON:
-		b, err := json.MarshalIndent(results, "", "    ")
-		if err != nil {
-			errFatalf("marshaling validation json results: %v", err)
+		var jsonB []byte
+		var err error
+		if stats != nil {
+			statsAndResults := map[string]interface{}{"results": results, "stats": stats}
+			jsonB, err = json.MarshalIndent(statsAndResults, "", fourSpaceTab)
+			if err != nil {
+				cmdutils.ErrFatalf("marshaling validation json results and stats: %v", err)
+			}
+		} else {
+			jsonB, err = json.MarshalIndent(results, "", fourSpaceTab)
+			if err != nil {
+				cmdutils.ErrFatalf("marshaling validation json results: %v", err)
+			}
 		}
-		return string(b)
+
+		return string(jsonB)
 	case stringYAML:
 		yamlResults := test.GetYamlFriendlyResults(results)
 		jsonb, err := json.Marshal(yamlResults)
 		if err != nil {
-			errFatalf("pre-marshaling results to json: %v", err)
+			cmdutils.ErrFatalf("pre-marshaling results to json: %v", err)
 		}
 
 		unmarshalled := []*test.YamlGatorResult{}
 		err = json.Unmarshal(jsonb, &unmarshalled)
 		if err != nil {
-			errFatalf("pre-unmarshaling results from json: %v", err)
+			cmdutils.ErrFatalf("pre-unmarshaling results from json: %v", err)
 		}
 
-		yamlb, err := yaml.Marshal(unmarshalled)
-		if err != nil {
-			errFatalf("marshaling validation yaml results: %v", err)
+		var yamlb []byte
+		if stats != nil {
+			statsAndResults := map[string]interface{}{"results": yamlResults, "stats": stats}
```

Comment:
The work done to `yamlResults` above (marshalling/unmarshalling) to create yamlb is not used here. That stuff should go in the other logic branch.

Reply:
Good point! Actually, when `stats == nil` that's not necessarily true but I've reworked the code as suggested since its cleaner and clearer!